02.calendar/cal.rb

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env ruby
+require 'date'
+require 'optparse'
+require 'debug'
+
+def main
+  year = Date.today.year
+  month = Date.today.month
+  if ARGV.any?
+    opt = OptionParser.new
+    opt.on("-y [YYYY]", "Specify a year") {|value| year = value.to_i}
+    opt.on("-m [MM]", "Specify a month") {|value| month = value.to_i}
+    opt.parse!
+  end
+  puts "      #{month}月 #{year}"
+  puts "日 月 火 水 木 金 土"
+  print_cal(year, month)
+end
+
+def print_cal(year, month)
+  first_day = Date.new(year, month, 1)
+  last_day = Date.new(year, month, -1)
+  print " ".rjust(first_day.wday*3) # 一行目のpadding
+  (first_day..last_day).each do |date|
+    puts "" if date.sunday? && date.day != 1 # 土曜日を表示した後に改行を入れる
+    print date.day < 10 ? "#{date.day}".ljust(3) : "#{date.day}".ljust(3) 
```

Comment:
trueとfalseの場合の処理が同じなので、こう書いても同じですね。

```suggestion
    print "#{date.day}".ljust(3)
```

ただ、この書き方でちゃんとカレンダーが表示されたのでしょうか？
最新の実行結果を提出物ページに追記してください〜。

Reply:
> まだ気になる点がいくつかあるのですが、ljustのメソッドの使い方は正しいでしょうか？

正しくない気がしますね。
まずはもう一度オリジナルのcalコマンドと自分のプログラムの実行結果を比較してみてください。
そうすれば何が悪いのか見えてくるんじゃないかな〜と思います。

Reply:
> 正しくない気がしますね。 まずはもう一度オリジナルのcalコマンドと自分のプログラムの実行結果を比較してみてください。 そうすれば何が悪いのか見えてくるんじゃないかな〜と思います。

結局、`center()`を使って欲しい見た目を手に入れた。この辺が、伊藤さんのお望みだったのかな？